server/gateway.go

```diff
@@ -2012,6 +2038,16 @@ func (s *Server) gatewayUpdateSubInterest(accName string, sub *subscription, cha
 		}
 	}
 	if first || last {
+		if first && sub.client != nil {
+			c := sub.client
+			if c.in.rsubs == nil {
+				c.in.rsubs = NewSublist()
```

Comment:
Why is this not hung off of gw?

Reply:
Yes, moving it there.

Reply:
Well actually no... we want this to be in a client connection's read cache because we want to detect that a connection has done a SUB and then a PUB with reply subject referencing a SUB that has been recently created (so will be a match in rsubs). Those client connections are NOT gateway connections, so gw will be nil.

Reply:
But this code only execs if GWs present correct? I am concerned about perf hit here.

Reply:
This code is invoked when gateways are enabled, yes, but we need to store the recent sub in the client's connection (the one doing the SUB followed by PUB), not the gw connections themselves.

Reply:
Let's run a benchmark regression.